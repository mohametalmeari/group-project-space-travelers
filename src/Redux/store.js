import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './features/missions/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
  },
});

export default store;
