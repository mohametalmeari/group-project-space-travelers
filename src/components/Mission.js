import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../Redux/features/missions/missionSlice';
import styles from '../styles/missions.module.css';

const Mission = ({
  id, name, description, reserved,
}) => {
  const dispatch = useDispatch();
  return (
    <tr className={styles.mission}>
      <td className={styles.nameTableCell}>{name}</td>
      <td className={styles.tableCell}>
        {description}
      </td>
      {
      !reserved && (
      <>
        <td className={styles.tableCell}>
          <span className={styles.notActiveStatus}>NOT A MEMBER</span>
        </td>
        <td className={styles.tableCell}>
          <button className={styles.joinBtn} type="button" onClick={() => dispatch(joinMission(id))}>
            Join Mission
          </button>
        </td>
      </>
      )
      }
      {
      reserved && (
      <>
        <td className={styles.tableCell}>
          <span className={styles.activeStatus}>Active Member</span>
        </td>
        <td className={styles.tableCell}>
          <button className={styles.leaveBtn} type="button" onClick={() => dispatch(leaveMission(id))}>
            Leave Mission
          </button>
        </td>
      </>
      )
      }

    </tr>
  );
};

Mission.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  id: PropTypes.string,
  reserved: PropTypes.bool,
};
export default Mission;
