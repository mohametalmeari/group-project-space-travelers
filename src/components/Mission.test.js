import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import Mission from './Mission';

describe('Mission Component', () => {
  test('Test render of Mission', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Mission />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
