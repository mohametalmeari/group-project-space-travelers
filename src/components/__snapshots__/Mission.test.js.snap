// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Component Test render of Mission 1`] = `
<div>
  <tr
    class="mission"
  >
    <td
      class="nameTableCell"
    />
    <td
      class="tableCell"
    />
    <td
      class="tableCell"
    >
      <span
        class="notActiveStatus"
      >
        NOT A MEMBER
      </span>
    </td>
    <td
      class="tableCell"
    >
      <button
        class="joinBtn"
        type="button"
      >
        Join Mission
      </button>
    </td>
  </tr>
</div>
`;
